packages/shared/src/features/onboarding/steps/FunnelQuiz.tsx

```diff
@@ -0,0 +1,95 @@
+import type { ReactElement } from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
+import type { FunnelStepQuiz } from '../types/funnel';
+import {
+  FunnelStepQuizQuestionType,
+  FunnelStepTransitionType,
+} from '../types/funnel';
+import { FormInputRating } from '../../common/components/FormInputRating';
+import { FormInputCheckboxGroup } from '../../common/components/FormInputCheckboxGroup';
+import ConditionalWrapper from '../../../components/ConditionalWrapper';
+import { FunnelStepCtaWrapper } from '../shared/FunnelStepCtaWrapper';
+
+const quizComponentsMap = {
+  [FunnelStepQuizQuestionType.Rating]: FormInputRating,
+  [FunnelStepQuizQuestionType.Radio]: FormInputCheckboxGroup,
+  [FunnelStepQuizQuestionType.Checkbox]: FormInputCheckboxGroup,
+};
+
+const checkIfSingleChoice = (type: FunnelStepQuizQuestionType): boolean => {
+  return (
+    type === FunnelStepQuizQuestionType.Radio ||
+    type === FunnelStepQuizQuestionType.Rating
+  );
+};
+
+export const FunnelQuiz = ({
+  id,
+  question,
+  onTransition,
+}: FunnelStepQuiz): ReactElement => {
+  const { type, text, options, imageUrl } = question;
+  const isSingleChoice = checkIfSingleChoice(type);
+  const [stepValue, setStepValue] = useState<string | string[]>([]);
+  const Component = useMemo(() => quizComponentsMap[type], [type]);
+  const inputOptions = useMemo(
+    () =>
+      options.map((option) => ({
+        label: option.label,
+        value: option.label,
```

Comment:
Is it expected to pass the label on both properties?
```suggestion
        value: option.value,
```

Reply:
Yes we should, but current schema from API do not have it. I'm going to update the schema today and then update this component before merging ✅